src/memory/memory.h

```diff
@@ -207,4 +213,37 @@ USE_RESULT bool memory_check_noise_remote_static_pubkey(const uint8_t* pubkey);
  */
 USE_RESULT bool memory_add_noise_remote_static_pubkey(const uint8_t* pubkey);
 
+typedef enum {
+    // success
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_OK,
+    // there was no more free slot available.
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_FULL,
+    // a different multisig with the same name already exists.
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_DUPLICATE_NAME,
+    // memory write error
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_UNKNOWN,
+} memory_multisig_set_by_hash_result_t;
```

Comment:
this type is very long. Couldn't you reuse the error codes within at least multisig? Why are there error codes unique to `SET_BY_HASH` ?

Reply:
I believe the goal is to have a single error type in the whole project, so at least try to have a single one in each module. So I think `memory_result_t` should be enough.

Reply:
Good point, done!